job/replay_manager_test.go

```diff
@@ -0,0 +1,91 @@
+package job_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/odpf/optimus/core/logger"
+	"github.com/odpf/optimus/job"
+	"github.com/odpf/optimus/mock"
+	"github.com/odpf/optimus/models"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReplayManager(t *testing.T) {
+	replayManagerConfig := job.ReplayManagerConfig{
+		NumWorkers:    5,
+		WorkerTimeout: 1000,
+	}
+	t.Run("Close", func(t *testing.T) {
+		logger.Init(logger.ERROR)
+		manager := job.NewManager(nil, nil, nil, replayManagerConfig)
+		err := manager.Close()
+		assert.Nil(t, err)
+	})
+	t.Run("Replay", func(t *testing.T) {
+		dagStartTime, _ := time.Parse(job.ReplayDateFormat, "2020-04-05")
+		startDate, _ := time.Parse(job.ReplayDateFormat, "2020-08-22")
+		endDate, _ := time.Parse(job.ReplayDateFormat, "2020-08-26")
+		jobSpec := models.JobSpec{
+			Name: "job-name",
+			Schedule: models.JobSpecSchedule{
+				StartDate: dagStartTime,
+				Interval:  "0 2 * * *",
+			},
+		}
+		replayRequest := &models.ReplayWorkerRequest{
+			Job:   jobSpec,
+			Start: startDate,
+			End:   endDate,
+			Project: models.ProjectSpec{
+				Name: "project-name",
+			},
+			JobSpecMap: map[string]models.JobSpec{
+				"job-name": jobSpec,
+			},
+		}
+		t.Run("should throw error if uuid provider returns failure", func(t *testing.T) {
+			logger.Init(logger.ERROR)
```

Comment:
change it to logger.InitWithWriter everywhere with ioutil.Discard and move it to top of the test file I guess?

Reply:
i checked for this line and replaced it, maybe I checked out the file changes